src/Ocelot/DownstreamRouteFinder/UrlMatcher/UrlPathPlaceholderNameAndValueFinder.cs

```diff
@@ -62,7 +57,7 @@ public Response<List<PlaceholderNameAndValue>> Find(string path, string query, s
                     }
                     else
                     {
-                        var placeholderValue = GetPlaceholderValue(pathTemplate, query, placeholderName, path, counterForPath + 1, '?');
+                        var placeholderValue = GetPlaceholderValue(pathTemplate, query, placeholderName, path, counterForPath, '?');
```

Comment:
Wow! How did you figure out this `+1` issue? 😉 
And what was the impact?

Reply:
When adding the query to unit tests I found that one of them was failing so tracing that I changed the plus 1 issue to handle specifically the no slash case only